app/src/common/shared/org/mozilla/vrbrowser/browser/LoginStorage.kt

```diff
@@ -0,0 +1,86 @@
+package org.mozilla.vrbrowser.browser
+
+import android.content.Context
+import android.util.Log
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.future.future
+import kotlinx.coroutines.launch
+import mozilla.components.concept.storage.Login
+import mozilla.components.lib.dataprotect.SecureAbove22Preferences
+import mozilla.components.lib.dataprotect.generateEncryptionKey
+import mozilla.components.service.fxa.SyncEngine
+import mozilla.components.service.fxa.sync.GlobalSyncableStoreProvider
+import mozilla.components.service.sync.logins.GeckoLoginStorageDelegate
+import mozilla.components.service.sync.logins.SyncableLoginsStorage
+import org.mozilla.vrbrowser.browser.components.GeckoLoginDelegateWrapper
+import org.mozilla.vrbrowser.browser.engine.EngineProvider
+import java.util.concurrent.CompletableFuture
+
+class LoginStorage(
+        val context: Context
+) {
+    /**
+     * Shared Preferences that encrypt/decrypt using Android KeyStore and lib-dataprotect for 23+
+     * only on Nightly/Debug for now, otherwise simply stored.
+     * See https://github.com/mozilla-mobile/fenix/issues/8324
+     */
+    private fun getSecureAbove22Preferences() =
+            SecureAbove22Preferences(
+                    context = context,
+                    name = KEY_STORAGE_NAME
+            )
+
+    private val passwordsEncryptionKey by lazy {
+        getSecureAbove22Preferences().getString(PASSWORDS_KEY)
+                ?: generateEncryptionKey(KEY_STRENGTH).also {
+                    if (SettingsStore.getInstance(context).isPasswordsEncryptionKeyGenerated) {
+                        // We already had previously generated an encryption key, but we have lost it
+                        Log.d(LOG_TAG,"Passwords encryption key for passwords storage was lost and we generated a new one")
+                    }
+                    SettingsStore.getInstance(context).recordPasswordsEncryptionKeyGenerated()
+                    getSecureAbove22Preferences().putString(PASSWORDS_KEY, it)
```

Comment:
Should we delete the value set in `getSecureAbove22Preferences` when logging out of FxA? I'm thinking on the use case of sharing a device with multiple people and see if there are secutity implications

Reply:
`getSecureAbove22Preferences` just generates a key to encrypt the logins storage so I don't think there are security implications in using the same key after an FxA logout?

Regarding the store lifecycle after an FxA logout, there is no way of selectively deleting only the synced logins after an FxA logout so we have to just wipe out all the login storage. There is a `wipeLocal` method that should do that but unfortunately doesn't work as expected and it just removes everything (https://github.com/mozilla-mobile/android-components/issues/8079). I guess removing everything is better than leaving logins after an FxA logout.